### PR TITLE
1.1.1

### PR DESCRIPTION
**Pull request by mertid**
1.1.1

### CODE REVIEW

File: tealium/__init__.py [hunk 1]
```diff
@@ -22,6 +22,14 @@ class Tealium(object):
     EVENT_TYPE_VIEW = "view"
     EVENT_TYPE_INTERACTION = "interaction"
 
+# argument labels for track call.
```
Comment (jalakoo) [inline tealium/__init__.py]: Lets list both the EVENT_TYPEs and TRACKs alphabetically.

File: tealium/__init__.py [hunk 2]
```diff
@@ -75,6 +84,15 @@ def getUUIDandSave(self):
             afile.close()
         return UUID
 
+    def isValidEventType(self, eventType):
+        eventTypeArray = [self.EVENT_TYPE_VIEW, self.EVENT_TYPE_DERIVED,
```
Comment (jalakoo) [inline tealium/__init__.py]: Break these up, one constant per line, and ordered alphabetically for clarity.

File: unittests.py [hunk 3]
```diff
@@ -34,7 +34,11 @@ def tealiumCallback(self, info, success, error=None):
     def test_trackEvent(self):
         t = Tealium('tealiummobile', 'demo', 'dev')
         d = {"foo": "bar"}
-        t.trackEvent('title', t.EVENT_TYPE_VIEW, d, self.tealiumCallback)
+        t.trackEvent(title="test",
```
Comment (jalakoo) [inline unittests.py]: Am I correct in assuming we can actually put these args in any order?

File: unittests.py [hunk 4]
```diff
@@ -34,7 +34,11 @@ def tealiumCallback(self, info, success, error=None):
...
+                     data=d,
+                     eventType=Tealium.EVENT_TYPE_VIEW,
+                     tealiumCallback=self.tealiumCallback)
+
         t = None
```
Comment (jalakoo) [inline unittests.py]: Presuming this is needed to clear out the variable?  If so, should be added to the other tests that assign a value to 't'

File: unittests.py (hunk 3 above)
Comment (mertid) [reply unittests.py]: yes

File: sample.py [hunk 5]
```diff
@@ -18,5 +18,7 @@ def tealiumCallback(info, success, error=None):
 
 t = Tealium('tealiummobile', 'demo', 'dev')
 d = {'testkey': 'testvalue'}
-t.trackEvent('title', Tealium.EVENT_TYPE_VIEW, d, tealiumCallback)
-t.trackEvent('title', d)
+t.trackEvent(title="title",
+             data=d,
+             eventType=Tealium.EVENT_TYPE_VIEW,
+             tealiumCallback=tealiumCallback)
```
Comment (ChadHartman) [inline sample.py]: Named arguments should only be for optional params.

File: tealium/__init__.py [hunk 6]
```diff
@@ -75,6 +85,17 @@ def getUUIDandSave(self):
             afile.close()
         return UUID
 
+    def isValidEventType(self, eventType):
+        eventTypeArray = [self.EVENT_TYPE_VIEW,
+                          self.EVENT_TYPE_DERIVED,
+                          self.EVENT_TYPE_ACTIVITY,
+                          self.EVENT_TYPE_CONVERSION,
+                          self.EVENT_TYPE_INTERACTION]
+        if (eventType in eventTypeArray):
```
Comment (ChadHartman) [inline tealium/__init__.py]: can be condensed into `return eventType in eventTypeArray`

File: tealium/__init__.py [hunk 7]
```diff
@@ -105,15 +126,16 @@ def enable(self):
             statusResult = False
         return statusResult
 
-    def trackEvent(self, title, eventType=None, data=None, tealiumCallback=None):
+    def trackEvent(self,  **arguments):
+
```
Comment (ChadHartman) [inline tealium/__init__.py]: the `**` argument is the ellipsis operator `...` in other languages. This seems to contradict what the docstring says. Is there a purpose for it?

Comment (mertid) [inline tealium/__init__.py]: From my understanding, that is the syntax for named parameters

**Pull request merged**